src/Adornment/Adornment.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { ReactNode } from 'react'
+
+import { AdornmentWrapper } from './styled'
+import { AdornmentContainer } from './AdornmentContainer'
+import { useAddWidthToPadding, useCloneChildWithPadding } from './hooks'
+
+export enum Position {
+  LEFT = 'Left',
+  RIGHT = 'Right'
+}
+
+export const ADORNMENT_DISPLAY_NAME = 'Adornment'
+export const EXTRA_PADDING = 'extraPadding'
+
+export interface ExtraPadding {
+  [Position.LEFT]: string
+  [Position.RIGHT]: string
+}
+
+type AdornmentProps = {
+  children: ReactNode
+  component: ReactNode
+
+  position?: Position
+  [EXTRA_PADDING]?: ExtraPadding
+}
+
+export const Adornment = ({
+  component,
+  position = Position.LEFT,
+  children,
+  [EXTRA_PADDING]: extraPadding = { [Position.RIGHT]: '0px', [Position.LEFT]: '0px' }
```

Comment:
I like how you set default values here! However will this not take away the left padding when you only have an adornment right?